### DISCUSSION TITLE
Mark query builder functions as const function

### DISCUSSION THREAD

**Discussion by Evian-Zhang** [SeaQL/sea-query#691] [Ideas]
## Motivation

I'm new to this crate. and during my experience of writing ORMs, I think it is wise to mark query builder functions as const functions. In most situations, one query corresponds to one static string, which we can define them in const environments. However, current implementation of query builders does not contain a `const` mark, while I didn't find something prevents doing so.

## Proposed Solutions

Mark query builder functions as const function.

**Comment by tyt2y3** [comment]
The current limitation of const functions in Rust would prevent this.
However, there could be other means to generate the queries compile time, probably by writing a very interesting `build.rs`

**Reply by Evian-Zhang** [reply]
After a quick thought, I think the build.rs solution may be vulnerable. Since the database manipulation is very sensitive, the potential attacker could also write a build.rs or proc macro to re-write the generated SQL sentence. So maybe a proc macro to produce SQL sentence could be more secure.
